Update ticdc canal json doc (#14943)
pingcap/docs: ticdc/ticdc-open-api-v2.md

| `encoder_concurrency`   | `INT` type. The number of encoder threads in the MQ sink. The default value is `16`. (Optional)                                                                                                                |
| `protocol`              | `STRING` type. For MQ sinks, you can specify the protocol format of the message. The following protocols are currently supported: `canal-json`, `open-protocol`, `canal`, `avro`, and `maxwell`.               |
| `schema_registry`       | `STRING` type. The schema registry address. (Optional)                                                                                                                                                         |
| `terminator`            | `STRING` type. The terminator is used to separate two data change events. The default value is null, which means `"\r\n"` is used as the terminator. (Optional)                                                |
| `transaction_atomicity` | `STRING` type. The atomicity level of the transaction. (Optional)                                                                                                                                              |

> | `only_output_updated_columns` | `BOOLEAN` type. For MQ sinks using the `canal-json` or `open-protocol` protocol, you can specify whether only output the modified columns. The default value is `false`. (Optional)  |

Reviewer: ```suggestion
| `only_output_updated_columns` | `BOOLEAN` type. For MQ sinks using the `canal-json` or `open-protocol` protocol, you can specify whether only output the modified columns. The default value is `false`.  |
```

Reviewer: (Optional) will be added in another PR https://github.com/pingcap/docs/pull/15046

> | `only_output_updated_columns` | `BOOLEAN` type，For MQ sinks, determines whether only output the modified columns. The following protocols are currently supported: `canal-json`, and `open-protocol`。The default value is `false`. (Optional)  |

Reviewer: ```suggestion
| `only_output_updated_columns` | `BOOLEAN` type. For MQ sinks using the `canal-json` or `open-protocol` protocol, you can specify whether only output the modified columns. The default value is `false`. (Optional)  |
```
